pkg/util/logging/persist_logger.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright The Velero Contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logging
+
+import (
+	"compress/gzip"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type PersistLogger interface {
+	logrus.FieldLogger
+	PersistLogStop(log logrus.FieldLogger)
+	GetPersistLogFile() (*os.File, error)
+	Dispose(log logrus.FieldLogger)
+}
+
+type persistLogger struct {
+	logrus.FieldLogger
+	logger *logrus.Logger
+	file   *os.File
+	w      *gzip.Writer
+	fields logrus.Fields
+}
+
+// NewPersistLogger creates the logger instance that writes logs to both Stdout and a file in the temp folder so that
+// it can be persisted later
+func NewPersistLogger(logLevel logrus.Level, logFormat Format, hook *LogHook, fields logrus.Fields) (PersistLogger, error) {
+	file, err := ioutil.TempFile("", "")
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating temp file")
+	}
+	w := gzip.NewWriter(file)
+
+	logger := DefaultLogger(logLevel, logFormat)
+	logger.Out = io.MultiWriter(os.Stdout, w)
+
+	if hook != nil {
+		logger.Hooks.Add(hook)
+	}
+
+	return &persistLogger{
+		FieldLogger: logger.WithFields(fields),
+		logger:      logger,
+		file:        file,
+		w:           w,
+		fields:      fields,
```

Comment:
I don't think the `fields` attribute is needed, b/c the logger in `PersisLogStop` is a field logger already, and the caller can set the fields properly it won't really miss useful information.

Reply:
Done, removed it